Dockerfile

```diff
@@ -3,11 +3,8 @@ FROM library/node:latest
 RUN mkdir /src
 WORKDIR /src
 
-COPY package.json .
-RUN npm install
-
 COPY . .
-RUN tools/prepare.sh
+RUN npm install
```

Comment:
```suggestion
RUN npm ci
```

Reply:
> npm install in the dockerfile triggers the prepare.sh script (As postinstall), which needs to be copied before the install starts. 

By the way, I always felt a bit uneasy about that `postinstall` script in `package.json`:
```
"postinstall": "./tools/prepare.sh",
```
It feels to me like abusing the scope of `npm install` (a more common use of the `postinstall` script is to apply patches to dependencies, which needs to happen before the app can be used, like we do in the `balena-cli` repo). The `prepare.sh` script in turn calls `fetch-external.sh` which then downloads the markdown files from more than 10 other repos. Doing this in `postinstall` catches developers by surprise, and makes it inconvenient to test with new dependencies and update them -- one would not want to trigger `fetch-external.sh` every time they run `npm install`.

So instead of deleting `tools/prepare.sh` from the Dockerfile, how about deleting the `postinstall` script from `package.json`? Then I think this Dockerfile could have:

```
COPY . .
RUN npm install
RUN tools/prepare.sh
```

"Explicit is better than implicit" - most of the time. :-)

Re `npm ci` vs `npm install`, it works too. Considerations:
* `npm ci` starts by deleting `node_modules`. Note that the `node_modules` folder is listed in the `.dockerignore` file (as it should be), so this step of deleting `node_modules` would be a no-op.
* `npm ci` does not modify the `package-lock.json` file, and requires `package-lock.json` to be consistent with the `pakcage.json` file (no conflicts). This is probably a good thing, but it also requires developers to actually commit changes to `package-lock.json`, which from some experience with the docs repo, developers had not been doing in the past. (I had also found in the past that Node 10 vs 12 vs 14 produced different changes to the `package-lock.json` file in this repo, but this is not really a reason to avoid `npm ci`.)

Reply:
Thanks @pdcastro @xginn8 I back `npm CI` too. The only issue with it is that package-lock.json needs to be updated as well. With our mutual understanding as codeowners, we can make sure that if someone does update the `package.json`, then `package-lock.json` needs to get committed as well. Doesn't happen that much tbh. 

I made this change in the PR to keep in line with the current structure. But, improving the workflow would be quite good. I agree with Paulo on deleting the `postinstall` in `npm install`. Though this also affects our contributor experience as folks would now need to 

1. npm install
2. Run a script --> That also triggers npm run build
3. npm start

Here's my recommendation, modify the `build` script to also run prepare.sh instead of the other way around. I will remove `npm run build` from `prepare.sh` script

```
"build": "./tools/prepare.sh && ./tools/build.sh && webpack -p",
```
Checkout the commit now.

Reply:
Would need to edit the README too, so don't merge this yet.